crates/ruff/src/rules/pylint/rules/global_statement.rs

```diff
@@ -0,0 +1,36 @@
+use ruff_macros::{define_violation, derive_message_formats};
+
+use crate::ast::types::BindingKind;
+use crate::checkers::ast::Checker;
+use crate::registry::Diagnostic;
+use crate::violation::Violation;
+
+define_violation!(
+    pub struct GlobalStatement {
+        pub name: String,
+    }
+);
+impl Violation for GlobalStatement {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let GlobalStatement { name } = self;
+        format!("Using the global statement to mutate `{name}` is discouraged")
+    }
+}
+
+// PLW0603
+pub fn global_statement(checker: &mut Checker, name: &str) {
```

Comment:
I think this can be a bit simpler. Above, I'd call `pylint::rules::global_statement(self, target, id)`, then make this `pub fn global_statement(checker: &mut Checker, expr: &Expr, name: &str)`, and then the body can just be:

```rust
checker.diagnostics.push(Diagnostic::new(
    GlobalStatement {
        name: name.to_string(),
    },
    Range::from_located(expr)
));
```